src/d3a/models/myco_matcher/__init__.py

```diff
@@ -0,0 +1,25 @@
+from d3a_interface.constants_limits import ConstSettings
+
+from d3a.constants import BidOfferMatchEnum
+from d3a.models.myco_matcher.pay_as_bid import PayAsBidMatch
+from d3a.models.myco_matcher.pay_as_clear import PayAsClear
+from d3a.d3a_core.exceptions import WrongMarketTypeException
+
+
+class MycoMatcher:
+    def __init__(self):
+        self.bid_offer_pairs = []
+        if ConstSettings.IAASettings.BID_OFFER_MATCH_TYPE == BidOfferMatchEnum.PAY_AS_BID.value:
+            self.match_algorithm = PayAsBidMatch()
+        elif ConstSettings.IAASettings.BID_OFFER_MATCH_TYPE == \
+                BidOfferMatchEnum.PAY_AS_CLEAR.value:
+            self.match_algorithm = PayAsClear()
+        else:
+            raise WrongMarketTypeException(f'Wrong market type setting flag '
+                                           f'{ConstSettings.IAASettings.MARKET_TYPE}')
+
```

Comment:
So, how do I integrate a custom matching algorithm now? Lets say I have a
```
class HannesMatcher(BaseMatcher):
```
I need to add it here, two, right? Wouldn't it be better to add a `CUSTOM` to the `BID_OFFER_MATCH_TYPE` enum ? Or is this not part of the story?

Reply:
Yes, to be extended as part of https://gridsingularity.atlassian.net/browse/D3ASIM-3447. For now restricted to only two types of double sided market to be sure if no loopholes are there at the moment of accidental introduction of any third market type.

Reply:
Thanks for explaining!